core/app/models/spree/zone.rb

```diff
@@ -58,6 +58,8 @@ def self.match(address)
     # current zone, if it's a state zone. If the passed-in zone has members, it
     # will also be in the result set.
     def self.with_shared_members(zone)
+      return none unless zone
```

Comment:
I had forgotten that `none` is an ActiveRecord method. Looked for our own implementation and didn't see one, and got confused. Sorry spreading the confusion.